packages/nimble/R/MCMC_samplers.R

```diff
@@ -284,44 +286,52 @@ sampler_RW_block_ETA <- nimbleFunction(
             returnType(double(1))
             return(propValueVector)
         },
-        adaptiveProcedure = function(jump = logical()) {
-            timesRan <<- timesRan + 1
+        adaptiveProcedure = function(jump = logical(), alpha = double()) {
+            LogProb         <- model$getLogProb()
+            timesRan       <<- timesRan + 1
+            EWMALogProb    <<- alpha * LogProb + (1-alpha) * EWMALogProb
+            maxEWMALogProb <<- max(maxEWMALogProb, EWMALogProb)
```

Comment:
It seems this is the line generating the compile error. It is the only line where I use "max()". Perhaps then the problem is with alpha (parameter for exponential weighted moving average), which I had tried to write as a user definable run-time parameter with a default.